From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Louis Scalbert <louis.scalbert@6wind.com>
Date: Thu, 13 Nov 2025 11:42:52 +0100
Subject: [PATCH] bgpd: do not add local vtep as remote

The previous commit has a side effect on "show evpn vni json".
The local VTEP is added to the list of remote VTEPs.

Do not add the local VTEP as remote.

Signed-off-by: Louis Scalbert <louis.scalbert@6wind.com>
(cherry picked from commit 5521d2e347af338a666fc2a2c5ae7a18b037c2e3)
Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
---
 bgpd/bgp_evpn.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bgpd/bgp_evpn.c b/bgpd/bgp_evpn.c
index 5d44d181fc..4302d1bb9a 100644
--- a/bgpd/bgp_evpn.c
+++ b/bgpd/bgp_evpn.c
@@ -925,6 +925,9 @@ static enum zclient_send_status bgp_zebra_send_remote_macip(
 	static struct ipaddr zero_remote_vtep_ip = { .ipa_type = IPADDR_V4, .ipaddr_v4 = { INADDR_ANY } };
 	bool esi_valid;
 
+	if (ipaddr_is_same(&vpn->originator_ip, remote_vtep_ip))
+		return ZCLIENT_SEND_SUCCESS;
+
 	/* Check socket. */
 	if (!bgp_zclient || bgp_zclient->sock < 0) {
 		if (BGP_DEBUG(zebra, ZEBRA))
